File: src/search/linearSearch.ts
import {ICanSearch} from "./ICanSearch";

export default class LinearSearch implements ICanSearch{
    search(array: number[], target: number): number {
        for(let i =0;i < array.length;i++){
            if(array[i] === target){
                return i;
            }
        }
        return -1;
    }
}

File: search/LinearSearch.java
package search;
public class LinearSearch implements ISearch{
    @Override
    public int search(int[] array, int target) {
        for(int i =0;i<array.length;i++){
            if(array[i] == target){
                return i;
            }
        }
        return -1;
    }
}